src/main/scala/viper/server/frontends/lsp/FileManager.scala

```diff
@@ -0,0 +1,382 @@
+
+  private def determineSuccess(code: Int): VerificationSuccess = {
+    if (diagnostics.isEmpty && code == 0) {
+      VerificationSuccess.Success
+    } else if (diagnostics.nonEmpty) {
+      //use tag and backend trace as indicators for completed parsing
+      if (!parsingCompleted) {
+        ParsingFailed
+      } else if (parsingCompleted && !typeCheckingCompleted) {
+        TypecheckingFailed
+      } else {
+        VerificationFailed
+      }
+    } else if (is_aborting) {
+      Aborted
+    } else {
+      Error
+    }
+  }
+
+  private def completionHandler(code: Int) {
+    try {
+      Log.debug(s"completionHandler is called with code ${code}")
+      if (is_aborting) {
+        is_verifying = false
+        return
+      }
+      var params: StateChangeParams = null
+      var success = NA
+
+      //      val isVerifyingStage = Coordinator.stage.getOrElse(return).isVerification
+      val isVerifyingStage = true
+
+      //do we need to start a followUp Stage?
+      if (isVerifyingStage) {
+        if (partialData != null && partialData.length > 0) {
+          Log.debug("Some unparsed output was detected:\n" + partialData)
+          partialData = ""
+        }
+
+        // Send the computed diagnostics to VSCode.
+        //Server.sendDiagnostics({ uri: this.fileUri, diagnostics: this.diagnostics })
+
+        //inform client about postProcessing
+        success = determineSuccess(code)
+        params = StateChangeParams(PostProcessing.id, filename = filename)
+        Coordinator.sendStateChangeNotification(params, Some(this));
+
+        //notify client about outcome of verification
+        val mt = if(this.manuallyTriggered) 1 else 0
+        params = StateChangeParams(Ready.id, success = success.id, manuallyTriggered = mt,
+          filename = filename, nofErrors = diagnostics.length, time = time.toDouble,
+          verificationCompleted = 1, uri = file_uri, error = internalErrorMessage)
+        Coordinator.sendStateChangeNotification(params, Some(this))
+
+        if (code != 0 && code != 1 && code != 899) {
+          Log.debug("Verification Backend Terminated Abnormally: with code " + code)
+        }
+      } else {
+        success = Success
+        val mt = if(this.manuallyTriggered) 1 else 0
+        params = StateChangeParams(Ready.id, success = success.id, manuallyTriggered = mt,
+          filename = filename, nofErrors = 0, time = time.toDouble,
+          verificationCompleted = 0, uri = file_uri, error = internalErrorMessage)
+        Coordinator.sendStateChangeNotification(params, Some(this))
+      }
+
+      //reset for next verification
+      lastSuccess = success
+      time = 0
+      is_verifying = false
+    } catch {
+      case e: Throwable =>
+        is_verifying = false
+        Coordinator.client.notifyVerificationNotStarted(file_uri)
+        Log.debug("Error handling verification completion: ", e)
+    }
+  }
+
+//  private def startVerificationTimeout(verificationCount: Int) = {
```

Comment:
What is the plan for this? Ideally, we shouldn't have any code that is commented out without explanations.

Reply:
No longer needed.